File: src/test/java/org/geepawhill/contentment/fragments/FaderTest.java
package org.geepawhill.contentment.fragments;

import static org.assertj.core.api.Assertions.*;

import org.geepawhill.contentment.core.*;
import org.junit.*;

import javafx.scene.Group;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

public class FaderTest
{
	private Group group;
	private Context context;

	@BeforeEach
	public void before()
	{
		group = new Group();
		context = new Context();
	}
	
	@Test
	public void fadesUp()
	{
		group.setOpacity(0d);
		Fader fader = new Fader(group,1d);
		fader.prepare(context);
		fader.interpolate(context, 1);
		assertThat(group.getOpacity()).isCloseTo(1d, within(0.01d));
	}
	
	@Test
	public void fadesDown()
	{
		group.setOpacity(1d);
		Fader fader = new Fader(group,0d);
		fader.prepare(context);
		fader.interpolate(context, 1);
		assertThat(group.getOpacity()).isCloseTo(0d, within(0.01d));
	}
	
	@Test
	public void worksPartially()
	{
		group.setOpacity(.25d);
		Fader fader = new Fader(group,1d);
		fader.prepare(context);
		fader.interpolate(context, .5d);
		assertThat(group.getOpacity()).isCloseTo(.625d, within(0.01d));
	}
}

File: src/test/kotlin/org/geepawhill/contentment/fragments/FaderTest.kt
package org.geepawhill.contentment.fragments

import javafx.scene.Group
import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.within
import org.geepawhill.contentment.core.Context
import org.junit.jupiter.api.Test

class FaderTest {
    private val group = Group()
    private val context = Context()

    @Test
    fun fadesUp() {
        group.opacity = 0.0
        val fader = Fader(group, 1.0)
        fader.prepare(context)
        fader.interpolate(context, 1.0)
        assertThat(group.opacity).isCloseTo(1.0, within(0.01))
    }

    @Test
    fun fadesDown() {
        group.opacity = 1.0
        val fader = Fader(group, 0.0)
        fader.prepare(context)
        fader.interpolate(context, 1.0)
        assertThat(group.opacity).isCloseTo(0.0, within(0.01))
    }

    @Test
    fun worksPartially() {
        group.opacity = .25
        val fader = Fader(group, 1.0)
        fader.prepare(context)
        fader.interpolate(context, .5)
        assertThat(group.opacity).isCloseTo(.625, within(0.01))
    }
}
